### DISCUSSION TITLE
View table loses type information

### DISCUSSION THREAD

**Discussion by DigNZ** [pocketbase/pocketbase#4277] [Q&A]
Hi,

I have a view table that gets the latest update timestamp for each connection. Unfortunately it loses the information that its a datetime and then the admin display shows it as an object or something instead. That's minor but does mean it loses the hover over conversion to local time which is a shame - also in my go code GetDateTime('latest_status_update') always returns the epoch instead of the datetime so it needs to be parsed. Is there something I'm doing wrong to get this information? Here is my view table query and the created is the usual record created field.

```
SELECT 
    c.id,
    c.name,
    c.connection_id,
    su.latest_status_update as latest_status_update,
    su.app_version
FROM 
    connections c
LEFT JOIN 
    (
        SELECT 
            connection, 
            MAX(created) AS latest_status_update,
            CAST(app_version as TEXT) as app_version
        FROM 
            status_update
        GROUP BY 
            connection, app_version
    ) su ON c.id = su.connection
WHERE 
    su.latest_status_update IS NOT NULL;
```

Edit GetDateTime already parses it but is still returning the epoch anyway. The field is interface{}(sting) with the correct datetime string in it but GetDateTime('latest_status_update') always returns the epoch.

**Comment by DigNZ** [comment]
Might be a red herring but the datetime value comes through with double quotes around it, though trimming (and setting it back) didn't help GetDateTime to return the time it showed.

**Comment by ganigeorgiev** [comment]
This is expected. Column expressions (columns from aggregations, subqueries, etc.) are always returned as `json` field type.

_Manual view schema mapping is [in the roadmap](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=32302192) but remains a low priority._
